build: Add Browserslist update on a schedule

rafaeelaudibert opened the pull request:
## What

Update our `browserslist` list on a schedule, every 10 days (give or take). This will run on GH and open a PR to allow us to review this.

## Does this work well for both Cloud and self-hosted?

Doesn't matter, it's only for our repo

## How did you test this code?

We can manually run the action if we want to, I'll test it that way

rafaeelaudibert committed:
build: Add Browserslist update on a schedule

rafaeelaudibert commented on .github/actions/browserslist-update/action.yml:
```diff
@@ -0,0 +1,31 @@
+                  # Setup for committing using built-in token. See https://github.com/actions/checkout#push-a-commit-using-the-built-in-token
+                  git config user.name "github-actions[bot]"
+                  git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            - name: Update Browserslist database and create PR if applies
+              uses: c2corg/browserslist-update-action@v2
+              with:
+                  base_branch: main
+                  commit_message: 'build: update Browserslist db'
+                  title: 'build: update Browserslist db'
+                  body: Auto-generated by [browserslist-update-action](https://github.com/c2corg/browserslist-update-action/)
```
Should we always assign someone to that PR to guarantee it gets reviewed eventually? What's the best team to assign this to?

pauldambra replied:
there's an automerge label that we use for posthog-js updates, you could use that

if CI passes and this is the only change we should be golden here

rafaeelaudibert replied:
Excellent call, let me add that real quick

rafaeelaudibert committed:
Automerge browserslist update PR

We have a label that can be used to automerge a PR in case all required tests are passing, let's use that for browserslist

pauldambra reviewed (approved):
LGTM
safe to get this in and fix forward if the robots need polishing

rafaeelaudibert commented on .github/actions/browserslist-update/action.yml:
```diff
@@ -0,0 +1,32 @@
+            - name: Update Browserslist database and create PR if applies
+              uses: c2corg/browserslist-update-action@v2
+              with:
+                  base_branch: main
```
No token, it uses the built-in one.

You did make me notice we should say `master` here rather than `main`, fixing it 🔧
